domain/ide/workspace/folder.go

```diff
@@ -150,63 +158,169 @@ func (f *Folder) ClearDiagnosticsFromPathRecursively(removedPath string) {
 	})
 }
 
+func (f *Folder) CheckAndUpdateStorage(path string, scannedIssues []snyk.Issue) {
+	method := "domain.ide.workspace.folder.CheckAndUpdateStorage"
+	cachedIssues, ok := f.documentDiagnosticCache.Load(path)
+	var productIssues []string
+
+	log.Debug().
+		Str("method", method).
+		Str("path", path).
+		Int("cachedIssuesCount", len(cachedIssues)).
+		Msg("Loaded issues from storage")
+
+	if ok && len(scannedIssues) == 0 {
+		f.documentDiagnosticCache.Delete(path)
+		return
+	}
+
+	if !ok || len(cachedIssues) == 0 {
+		f.documentDiagnosticCache.Store(path, scannedIssues)
+		return
+	}
+
+	// Map of issue IDs for quick lookup
+	cachedIssuesMap := make(map[string]snyk.Issue)
+	for _, issue := range cachedIssues {
+		cachedIssuesMap[issue.ID] = issue
+	}
+
+	// Check if each scanned issue is in the cache
+	for _, scanned := range scannedIssues {
```

Comment:
question: what happens if there are old findings in the cache, that are fixed and not present in the current scan? How are they removed?